lib/matplotlib/axes/_axes.py

```diff
@@ -2525,10 +2525,16 @@ def broken_barh(self, xranges, yrange, **kwargs):
         self._process_unit_info(xdata=xdata,
                                 ydata=ydata,
                                 kwargs=kwargs)
-        xranges = self.convert_xunits(xranges)
+        xnew = []
+        for xr in xranges:
+            # convert the absolute values, not the x and dx...
+            xnew.append([self.convert_xunits(xr[0]),
+                         self.convert_xunits(xr[0]+xr[1])])
+            xnew[-1][1] = xnew[-1][1] - xnew[-1][0]
```

Comment:
I feel a temporary variable might be a bit clearer than post-modifying the list:
```suggestion
            x0 = self.convert_xunits(xr[0])
            xnew.append([x0, self.convert_xunits(xr[0] + xr[1]) - x0])
```

Reply:
Yep, done...